portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,26 +38,27 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final Translate translate = TranslateOptions.getDefaultInstance().getService();
   private static final UserService userService = UserServiceFactory.getUserService();
   private static final String ENTITY_KIND = "Comment";
   private static final String ENTITY_NAME_HEADER = "name";
   private static final String ENTITY_EMAIL_HEADER = "email";
   private static final String ENTITY_COMMENT_HEADER = "comment";
   private static final String ENTITY_TIMESTAMP_HEADER = "timestamp";
+  private static String currentLanguage = "en";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    System.out.println(request.getParameter("lang"));
-
     List<String> comments = new ArrayList<>();
     Query query = new Query(ENTITY_KIND).addSort(ENTITY_TIMESTAMP_HEADER, SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
-
+    String languageCode = request.getParameter("lang");
     for (Entity entity : results.asIterable()) {
-      comments.add(buildComment(entity));
+      comments.add(buildComment(entity, languageCode));
     }
-
+    currentLanguage = languageCode;
```

Comment:
This may have issue. If two users send two requests, one choose Chinese and then second people choose Chinese, it will show English for the second person right? The translation depends on pre-request language selection, not on the text itself.

Reply:
Fixed.